### PR TITLE
Migrate TimeField component to Typescript

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
## Summary

Migrate [TimeField](https://uikit.commercetools.com/?path=/story/components-fields--timefield) and all of its subcomponent to Typescript.

### CODE REVIEW

File: packages/components/fields/time-field/src/time-field.tsx [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+const sequentialId = createSequentialId('time-field-');
+
+const hasErrors = (errors?: TFieldErrors) =>
+  errors && Object.values(errors).some(Boolean);
+
+class TimeField extends Component<TTimeFieldProps, TTimeFieldState> {
+  static displayName = 'TimeField';
+
+  static defaultProps = {
```
Comment (Rhotimee) [inline packages/components/fields/time-field/src/time-field.tsx:188]: It'll be nice to type the defaultProps.

**Review by Rhotimee — COMMENTED** [review]
Generally looks good. Just a minor feedback

File: packages/components/fields/time-field/src/time-field.tsx (hunk 1 above)
Comment (CarlosCortizasCT) [reply packages/components/fields/time-field/src/time-field.tsx]: Typed with this commit: 9990fac4c1e0c0f9c7f1d6bedde91885131541e3

File: packages/components/fields/time-field/src/time-field.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+};
+
+type TFieldErrors = Record<string, boolean>;
+
+type TTimeFieldProps = {
+  // TimeField
+  /**
+   * Used as HTML id property. An id is auto-generated when it is not specified.
+   */
+  id: string;
```
Comment (Rhotimee) [inline packages/components/fields/time-field/src/time-field.tsx:37]: I think this should be optional

Comment (CarlosCortizasCT) [reply packages/components/fields/time-field/src/time-field.tsx]: you are so right

Fixed in 029c99d7583c8d4d2b365acea64cbce037dfb360

Thanks 👍

File: packages/components/fields/time-field/src/time-field.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+
+  /**
+   * Called when input is blurred
+   */
+  onBlur?: (event: TEvent) => void;
+
+  /**
+   * Called when input is focused
+   * <br/>
+   * Signature: `(event) => void`
```
Comment (kark) [inline packages/components/fields/time-field/src/time-field.tsx:115]: Nit: if I understand correctly we have that comment autogenerated now

Comment (CarlosCortizasCT) [reply packages/components/fields/time-field/src/time-field.tsx]: Fixed in 7df74c0512fbc91a527c94a3daf6f1563f2cb405

Thanks for letting me know about this one 👍

**Review by Rhotimee — APPROVED** [review]

**Review by kark — APPROVED** [review]

**Review by ddouglasz — APPROVED** [review]
LGTM

**Pull request merged**